server/shadow/Sample/sample_shadow.c

```diff
@@ -0,0 +1,177 @@
+                                            WINPR_ATTR_UNUSED rdpShadowClient* client,
+                                            WINPR_ATTR_UNUSED UINT16 flags,
+                                            WINPR_ATTR_UNUSED UINT16 x, WINPR_ATTR_UNUSED UINT16 y)
+{
+	WLog_WARN(TAG, "TODO: Implement!");
+	return TRUE;
+}
+
+static BOOL sample_shadow_input_extended_mouse_event(
+    WINPR_ATTR_UNUSED rdpShadowSubsystem* subsystem, WINPR_ATTR_UNUSED rdpShadowClient* client,
+    WINPR_ATTR_UNUSED UINT16 flags, WINPR_ATTR_UNUSED UINT16 x, WINPR_ATTR_UNUSED UINT16 y)
+{
+	WLog_WARN(TAG, "TODO: Implement!");
+	return TRUE;
+}
+
+static UINT32 sample_shadow_enum_monitors(WINPR_ATTR_UNUSED MONITOR_DEF* monitors,
+                                          WINPR_ATTR_UNUSED UINT32 maxMonitors)
+{
+	WLog_WARN(TAG, "TODO: Implement!");
+	return 0;
+}
+
+static int sample_shadow_subsystem_init(rdpShadowSubsystem* arg)
+{
+	sampleShadowSubsystem* subsystem = (sampleShadowSubsystem*)arg;
+	WINPR_ASSERT(subsystem);
+
+	subsystem->base.numMonitors = sample_shadow_enum_monitors(subsystem->base.monitors, 16);
+
+	WLog_WARN(TAG, "TODO: Implement!");
+
+	MONITOR_DEF* virtualScreen = &(subsystem->base.virtualScreen);
+	virtualScreen->left = 0;
+	virtualScreen->top = 0;
+	virtualScreen->right = 0;
+	virtualScreen->bottom = 0;
+	virtualScreen->flags = 1;
+	return 1;
+}
+
+static int sample_shadow_subsystem_uninit(rdpShadowSubsystem* arg)
+{
+	sampleShadowSubsystem* subsystem = (sampleShadowSubsystem*)arg;
+
+	if (!subsystem)
+		return -1;
+
+	WLog_WARN(TAG, "TODO: Implement!");
+	return 1;
+}
+
+static int sample_shadow_subsystem_start(rdpShadowSubsystem* arg)
+{
+	sampleShadowSubsystem* subsystem = (sampleShadowSubsystem*)arg;
+
+	if (!subsystem)
+		return -1;
+
+	WLog_WARN(TAG, "TODO: Implement!");
+
+	return 1;
+}
+
+static int sample_shadow_subsystem_stop(rdpShadowSubsystem* arg)
+{
+	sampleShadowSubsystem* subsystem = (sampleShadowSubsystem*)arg;
+
+	if (!subsystem)
+		return -1;
+
+	WLog_WARN(TAG, "TODO: Implement!");
+
+	return 1;
+}
+
+static void sample_shadow_subsystem_free(rdpShadowSubsystem* arg)
+{
+	sampleShadowSubsystem* subsystem = (sampleShadowSubsystem*)arg;
+
+	if (!subsystem)
+		return;
+
+	sample_shadow_subsystem_uninit(arg);
+	free(subsystem);
+}
+
+static rdpShadowSubsystem* sample_shadow_subsystem_new(void)
+{
+	sampleShadowSubsystem* subsystem =
+	    (sampleShadowSubsystem*)calloc(1, sizeof(sampleShadowSubsystem));
+
+	if (!subsystem)
+		return NULL;
+
+	subsystem->base.SynchronizeEvent = sample_shadow_input_synchronize_event;
+	subsystem->base.KeyboardEvent = sample_shadow_input_keyboard_event;
+	subsystem->base.UnicodeKeyboardEvent = sample_shadow_input_unicode_keyboard_event;
+	subsystem->base.MouseEvent = sample_shadow_input_mouse_event;
+	subsystem->base.ExtendedMouseEvent = sample_shadow_input_extended_mouse_event;
+	return &subsystem->base;
+}
+
+const char* ShadowSubsystemName(void)
```

Comment:
`extern "C"` missing?

Reply:
not required, already declared in header